Move imports in nuheat component

djpremier opened the pull request:
## Description:
Moved imports from functions to top-level as requested in #27284

## Checklist:
  - [ ] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

djpremier committed:
Move imports in nuheat component

cgtobi commented:
Please fix the tests as well.

djpremier commented:
Ok, it will be done

djpremier commented:
@cgtobi The errors of this test (at least what is preventing it from advancing now), is the incorrect username / password to connect to the Nuheat api. I have no account, so I can't fix it.

djpremier committed:
Fix tox tests

djpremier commented:
Required module is commented in `requirement_all.txt`

balloob commented on tests/components/nuheat/test_init.py:
```diff
@@ -27,11 +27,12 @@ def tearDown(self):  # pylint: disable=invalid-name
     @patch("homeassistant.helpers.discovery.load_platform")
     def test_setup(self, mocked_nuheat, mocked_load):
         """Test setting up the NuHeat component."""
+        nuheat.nuheat = mocked_nuheat
```
This is not good. Your mock will never be reverted and impact other tests. Use this instead:

```suggestion
        with patch.object(nuheat, 'nuheat', mocked_nuheat):
            …
```

MartinHjelmare replied:
This was done in a couple of other PRs recently merged too. It would be good to fix those.

@djpremier probably knows which  integrations needed tests adjusting.

djpremier replied:
Hi Martin, I did, as far as I can remember was more `melissa`, another one that needed` wake_on_lan`, but this one is with @fabaff, and I already commented on the correct form for it. But as soon as possible I analyze the tests and do the PR. Thanks.

MartinHjelmare replied:
Yes, those are the ones. Thanks!

djpremier committed:
Fix tox tests

djpremier committed:
Update tests/components/nuheat/test_init.py

@Balloob suggested the change because direct replacement, the mock would never be reverted and impact the other tests.

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>
